src/common/task_system/task.cpp

```diff
@@ -23,6 +23,7 @@ void Task::deRegisterThreadAndFinalizeTask() {
         } catch (std::exception& e) { setExceptionNoLock(std::current_exception()); }
     }
     if (isCompletedNoLock()) {
+        updateProgress();
```

Comment:
As we discussed, this may not be a good place to update progress. So let's refresh each time we update but only if an amount of work (say 1%) has been finished since last refresh.